app/models/spree/slide.rb

```diff
@@ -1,19 +1,24 @@
 class Spree::Slide < ActiveRecord::Base
-
   has_and_belongs_to_many :slide_locations,
                           class_name: 'Spree::SlideLocation',
                           join_table: 'spree_slide_slide_locations'
 
-  has_attached_file :image,
-                    url: '/spree/slides/:id/:style/:basename.:extension',
-                    path: ':rails_root/public/spree/slides/:id/:style/:basename.:extension',
-                    convert_options: { all: '-strip -auto-orient -colorspace sRGB' }
-  validates_attachment :image, content_type: { content_type: ["image/jpg", "image/jpeg", "image/png", "image/gif"] }
+  belongs_to :product, touch: true, optional: true
+
+  has_one_attached :image
+
+  validates :name, :link_url, :image, presence: true, unless: -> { product }
```

Comment:
could we add validation for `link_url` to verify if url have proper format ?

Reply:
url validator has been added